contributors/design-proposals/local-storage-overview.md

```diff
@@ -0,0 +1,381 @@
+2. The cluster admin creates a DaemonSet addon that discovers all the raw block devices on a node that are available within a specific directory and creates corresponding PVs for them with a ‘StorageType’ of ‘Block’
+
+    ```yaml
+    kind: PersistentVolume
+    apiVersion: v1
+    metadata:
+      name: foo
+      annotations:
+        storage.kubernetes.io/node: node-1
+      labels:
+        storage.kubernetes.io/medium: ssd
+    spec:
+      volumeType: local
+      storageLevel: block
+      capacity:
+        storage: 100Gi
+      hostpath:
+        path: /var/lib/kubelet/storage-raw-devices/foo
+      accessModes:
+        - ReadWriteOnce
+      persistentVolumeReclaimPolicy: Delete
+    ```
+
+3. Bob creates a pod with a PVC that requests for block level access and similar to a Stateful Set scenario the scheduler will identify nodes that can satisfy the pods request.
+
+    ```yaml
+    kind: PersistentVolumeClaim
+    apiVersion: v1
+    metadata:
+      name: myclaim
+      labels:
+        storage.kubernetes.io/medium: ssd
+    spec:
+      volumeType: local
+      storageLevel: block
+      accessModes:
+        - ReadWriteOnce
+      resources:
+        requests:
+          storage: 80Gi
+    ```
+
+*The lifecycle of the block level PV will be similar to that of the scenarios explained earlier.* 
+
+# Open Questions & Discussion points 
+* Single vs split “limit” for storage across writable layer and logs
+* Local Persistent Volume bindings happening in the scheduler vs in PV controller
+    * Should the PV controller fold into the scheduler
+* Supporting dedicated partitions for logs and volumes in Kubelet in addition to runtime overlay filesystem
+    * This complicates kubelet.Not sure what value it adds to end users.
+* Providing IO isolation for ephemeral storage
+    * IO isolation is difficult. Use local PVs for performance
+* Support for encrypted partitions
+* Do applications need access to performant local storage for ephemeral use cases? Ex. A pod requesting local SSDs for use as ephemeral scratch space.
+    * Typically referred to as “inline PVs” in kube land
+* Should block level storage devices be auto formatted to be used as file level storage?
+    * Flexibility vs complexity
+    * Do we really need this? 
+* Repair/replace scenarios. 
+    * What are the implications of removing a disk and replacing it with a new one? 
+    * We may not do anything in the system, but may need a special workflow
+
+# Recommended Storage best practices
+
+* Have the primary partition on a reliable storage device
+* Consider using RAID and SSDs (for performance)
+* Partition the rest of the storage devices based on the application needs
+    * SSDs can be statically partitioned and they might still meet IO requirements of apps.
+    * TODO: Identify common durable storage requirements for most databases
+* Avoid having multiple logical partitions on hard drives to avoid IO isolation issues
```

Comment:
As it stands today, this solution is not changing primary partition IO isolation semantics, and doesn't increase operator cost for managing the primary partition.

For local persistent use cases, the only way to get IO isolation today is to use HostPath volumes, and data gravity with node affinity, all of which already has a high operator cost.  I believe this solution is an improvement on this existing method and will decrease operator cost for both the cluster admin and the application developer.  But if you see any ways that operator cost is increasing compared to the current methods, please let me know.

1) Dev and admin need to coordinate less about exact nodes and storage paths (but still has some coordination about number of disks and capacity requirements).  Operator costs greatly reduced because you don't have to manually schedule your pods on nodes using node affinity.
2) Pods are more flexible about where they can run because they don't need to specify node affinity.  Node failure scenarios can automatically be handled via forgiveness, instead of requiring manual intervention.
3) In order to use a pod template with HostPath, your nodes have to be provisioned the same way.  This proposal allows for heterogeneous node storage configs, which can make upgrading storage capacity easier for the admin.
4) With HostPath volumes, the admin has to manually cleanup the volume to be able to reuse it.  This proposal includes a workflow for automatically cleaning up and recycling the PVs.  This will greatly benefit multi-tenant clusters in the future.